Backend/api/workout/model.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require("mongoose");
+
+// Create a schema (blueprint) for workouts
+const workoutSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+    minlength: 8,
+  },
+  date: {
+    type: Date,
+    required: true,
+    min: Date.now,
+  },
+  type: {
+    type: String,
+    required: true,
+  },
+  exercises: {
+    type: [
+      {
+        name: {
+          type: String,
+          required: true,
+        },
+        sets: {
+          type: Number,
+          required: true,
+        },
+        reps: {
+          type: Number,
+          required: true,
+        },
+        rest: {
+          type: Number,
+          required: true,
+        },
+        weight: {
+          type: Number,
+          default: "No specific weight.",
```

Comment:
This is a UI specific string. You can either use 0 as a default value, or just leave it unset.